demo/repo/Jenkinsfile

```diff
@@ -1,15 +1,8 @@
-def splits = splitTests parallelism: count(5), generateInclusions: true
-def branches = [:]
-for (int i = 0; i < splits.size(); i++) {
-  def split = splits[i]
-  branches["split${i}"] = {
-    node('!master') {
-      checkout scm
-      writeFile file: (split.includes ? 'inclusions.txt' : 'exclusions.txt'), text: split.list.join("\n")
-      writeFile file: (split.includes ? 'exclusions.txt' : 'inclusions.txt'), text: ''
-      sh 'mvn -s /tmp/settings.xml -B clean test -Dmaven.test.failure.ignore'
-      junit 'target/surefire-reports/*.xml'
-    }
+@Library('testInParallel') _
+testInParallel(count(5), 'inclusions.txt', 'exclusions.txt', 'target/surefire-reports/TEST-*.xml', 'maven:3.3.9-jdk-8', {
+  checkout scm
```

Comment:
Also works to start the script with

```groovy
node {
  checkout scm
  stash 'sources'
}
```

and then have this `prepare` block just be

```groovy
unstash 'sources'
```

so that only one checkout is done per build. Unfortunately allocating agents using the Mock Slave plugin is pretty slow so this adds ~20s to the build.